### PR TITLE
Update in-product docs links to subscriptions documentation.

### PR DESCRIPTION
**Pull request by Brianmitchtay**
In the product data area there is a "Learn More" link to the docs. After some recent work in the documentation, this link was no longer resolving. This PR updates the old link from 
- http://docs.woocommerce.com/document/subscriptions/store-manager-guide/#limit-subscription 
to 
- https://woocommerce.com/document/subscriptions/creating-subscription-products/#limit-subscriptions

CleanShot 2024-08-08 at 09 50 17@2x

After some discussion with @james-allan I went ahead and updated all old references to docs.woocommerce.com that were redirecting (most still to the correct documentation) so they point to the correct places now.

Fixes #

## Description

This PR mostly just changes in-product doc link references. I also removed a message asking users to submit login credentials for their site when opening tickets about PayPal Payments IPN errors ([commit])(https://github.com/Automattic/woocommerce-subscriptions-core/pull/650/commits/128fea5c3bc5aa392b1967f06d99911455eb598b))

In some cases PHPCS was failing after making changes to only the urls, in these cases I resolved errors with my best effort, including breaking arguments from multi-line function calls on to their own lines, and adding an escape HTML function in [this commit](https://github.com/Automattic/woocommerce-subscriptions-core/pull/650/commits/caf79beb0d77064e712874e898820734320cb3be) which should be reviewed for correctness.

## How to test this PR

1. Create a new subscription product
2. Go to the "Advanced" tab of the product data area
3. Hover over the "Learn More" link to check it points to https://woocommerce.com/document/subscriptions/creating-subscription-products/#limit-subscriptions
4. Click the "Learn More" link and confirm it resolves to https://woocommerce.com/document/subscriptions/creating-subscription-products/#limit-subscriptions

## Product impact

- [x] Added changelog entry (or does not apply) No Changelog entry needed
- [x] NO -  Will this PR affect WooCommerce Subscriptions? yes/no/tbc, add issue ref
- [x] NO - Will this PR affect WooCommerce Payments? yes/no/tbc, add issue ref
- [x] None  Added deprecated functions, hooks or classes to the [spreadsheet](https://docs.google.com/spreadsheets/d/1xw9xszcPMnWsp4C8OKZMsLzZob7tOmWT7qMqmEIq314/edit#gid=0)

### CODE REVIEW

**Comment by Brianmitchtay** [comment]
@a-danae after talking with @james-allan I added the additional link updates to this same branch/PR. I hope that's alright. Please ping me if you have any questions.

File: includes/admin/meta-boxes/views/html-unknown-related-orders-row.php [hunk 1]
```diff
@@ -18,10 +18,17 @@
 		echo sprintf( esc_html_x( '#%s', 'hash before order number', 'woocommerce-subscriptions' ), esc_html( $order_id ) );
 		?>
 		<div class="wcs-unknown-order-info-wrapper">
-			<a href="https://docs.woocommerce.com/document/subscriptions/orders/#section-8">
+			<a href="https://woocommerce.com/document/subscriptions/orders/#why-are-some-orders-in-the-related-orders-table-not-linking-to-the-order">
 				<?php
-				// Translators: The %1 placeholder is the translated order relationship ("Parent Order"), %2 placeholder is a <br> HTML tag.
-				echo wcs_help_tip( sprintf( __( 'This %1$s couldn\'t be loaded from the database. %1$s Click to learn more.', 'woocommerce-subscriptions' ), $relationship, '</br>' ) ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+				echo esc_html(
```
Comment (a-danae) [inline includes/admin/meta-boxes/views/html-unknown-related-orders-row.php]: How about removing this esc_html() call? 

It's escaping the HTML that's intended to be printed. The function [wcs_help_tip()](https://github.com/Automattic/woocommerce-subscriptions-core/blob/trunk/includes/wcs-compatibility-functions.php#L18-L37) already escapes it via [wc_help_tip()](https://hooks.wbcomdesigns.com/reference/functions/wc_help_tip/). We _may_ want to escape the `$relationship` variable alone.

| With escaping | Without escaping |
|--------|--------|
| image | image | 

We could then add the comment `// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped`  above the `echo` statement to silence phpcs' warning.

**Review by a-danae — CHANGES_REQUESTED** [review]
Hey @Brianmitchtay 👋  thanks for updating the URLs in the Subscriptions core!

I'm requesting changes for an esc_html call for which I left a comment, but everything else looks good 🙂 

Thanks again for taking the time to work on this!

File: includes/admin/meta-boxes/views/html-unknown-related-orders-row.php (hunk 1 above)
Comment (Brianmitchtay) [reply includes/admin/meta-boxes/views/html-unknown-related-orders-row.php]: Thanks Danae, I'd had a feeling I was getting something wrong on that one. I appreciate you setting me right. It looks like the comment was already in the doc, but me breaking the function to multiple lines caused it to not get recognized at the right time by phpcs. Should be fixed now. 🤞

**Review by Brianmitchtay — COMMENTED** [review]
Changes made as requested!

**Comment by Brianmitchtay** [comment]
@a-danae forgive me, I'm not familiar with the review process. I've made the changes you requested and all tests are passing. Are you the one now who needs to approve/merge the PR? I've resolved the conversation from your review but I still see that changes are requested.

Thanks for your help 🙇‍♂️

**Review by james-allan — APPROVED** [review]
@Brianmitchtay changes look good. Thanks again for the PR. I'll merge it so you don't have to return here. :)

**Pull request merged**